Library/Homebrew/cask/lib/hbc/cask.rb

```diff
@@ -85,6 +85,15 @@ def installed?
       !versions.empty?
     end
 
+    def latest_installed_version
+      versions.last
+    end
+
+    def outdated?(greedy: false)
+      (greedy || !auto_updates && !version == :latest) &&
+      !version.nil? && (version.casecmp latest_installed_version) == 1
```

Comment:
I agree. More the point is outdated should mean "does not match the version in HBC"